service/history/chasm_notifier.go

```diff
@@ -0,0 +1,178 @@
+// This file implements ChasmNotifier, which allows subscribers to subscribe to notifications
+// relating to components in a specified CHASM execution. It is based on the events.Notifier
+// implementation.
+package history
+
+import (
+	"sync/atomic"
+	"time"
+
+	"github.com/dgryski/go-farm"
+	"github.com/google/uuid"
+	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/server/chasm"
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/clock"
+	"go.temporal.io/server/common/collection"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/service/history/configs"
+)
+
+type (
+	// ChasmNotifier allows subscribers to subscribe to notifications relating to components in a
+	// specified CHASM execution.
+	ChasmNotifier struct {
+		timeSource      clock.TimeSource
+		metricsHandler  metrics.Handler
+		status          int32
```

Comment:
Use the more modern https://pkg.go.dev/sync/atomic#Int32 instead.

Reply:
Thanks, I've now changed the PR to have a CHASM-specific notifier with a much simpler implementation than the workflow one. There is no longer a `status` field.